app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb

```diff
@@ -0,0 +1,73 @@
+class ManageIQ::Providers::IbmCloud::Inventory::Parser::VPC < ManageIQ::Providers::IbmCloud::Inventory::Parser
+  require_nested :CloudManager
+
+  attr_reader :img_to_os
+
+  def initialize
+    @img_to_os = {}
+  end
+
+  def parse
+    images
+    instances
+  end
+
+  def images
+    collector.images.each do |image|
+      img_to_os[image[:id]] = image&.dig(:operating_system, :name)
+      persister.miq_templates.build(
+        :uid_ems            => image[:id],
+        :ems_ref            => image[:id],
+        :name               => image[:name],
+        :description        => image&.dig(:operating_system, :display_name),
+        :location           => collector.manager.provider_region,
+        :vendor             => "ibm",
+        :connection_state   => "connected",
+        :raw_power_state    => "never",
+        :template           => true,
+        :publicly_available => true
+      )
+    end
+  end
+
+  def instances
+    collector.vms.each do |instance|
+      persister_instance = persister.vms.build(
+        :description      => "IBM Cloud Server",
+        :ems_ref          => instance[:id],
+        :location         => instance&.dig(:zone, :name),
```

Comment:
`:location` has a `validates_presence_of` so if it `instance&.dig(:zone, :name` can be `nil` you probably want to `instance&.dig(:zone, :name) || "unknown"`